### PR TITLE
Update track_total_hits to union type

### PR DESCRIPTION
**Pull request by russcam**
This commit updates track_total_hits parameter type to a union
of boolean and number, to reflect the possible values that can
be passed.

### CODE REVIEW

File: rest-api-spec/src/main/resources/rest-api-spec/api/search.json [hunk 1]
```diff
@@ -182,8 +182,8 @@
         "description":"Whether to calculate and return scores even if they are not used for sorting"
       },
       "track_total_hits":{
-        "type":"boolean",
-        "description":"Indicate if the number of documents that match the query should be tracked"
+        "type":"boolean|number",
+        "description":"Indicate if the number of documents that match the query should be tracked. A number can also be specified, to accurately track the total hit count up to the number"
```
Comment (karmi) [inline rest-api-spec/src/main/resources/rest-api-spec/api/search.json]: Missing period in the second sentence?

Comment (russcam) [reply rest-api-spec/src/main/resources/rest-api-spec/api/search.json]: I'm not seeing it - could you add as a suggestion?

Comment (karmi) [inline rest-api-spec/src/main/resources/rest-api-spec/api/search.json]:
```suggestion
        "description":"Indicate if the number of documents that match the query should be tracked. A number can also be specified, to accurately track the total hit count up to the number."
```

Comment (russcam) [reply rest-api-spec/src/main/resources/rest-api-spec/api/search.json]: thanks!

File: rest-api-spec/src/main/resources/rest-api-spec/api/search.json [hunk 2]
```diff
@@ -182,8 +182,8 @@
         "description":"Whether to calculate and return scores even if they are not used for sorting"
       },
       "track_total_hits":{
-        "type":"boolean",
-        "description":"Indicate if the number of documents that match the query should be tracked"
+        "type":"boolean|number",
```
Comment (russcam) [inline rest-api-spec/src/main/resources/rest-api-spec/api/search.json]:
```suggestion
        "type":"boolean|long",
```

This should be `long`, in line with https://github.com/elastic/elasticsearch/tree/master/rest-api-spec#type-definition

**Comment by colings86** [comment]
@russcam are you still wanting to progress this PR or should it be closed?

**Comment by russcam** [comment]
Yes I am @colings86. If it LGTY, I'll merge and backport

**Comment by colings86** [comment]
@jimczi could you work with @russcam on this change since its regarding track total hits?

**Comment by jimczi** [comment]
@elasticmachine update branch

**Review by jimczi — APPROVED** [review]
Sorry for the late review, LGTM

**Pull request merged**